rundeckapp/src/main/groovy/com/dtolabs/rundeck/app/tree/TreeUpdater.groovy

```diff
@@ -0,0 +1,74 @@
+package com.dtolabs.rundeck.app.tree
+
+import com.dtolabs.rundeck.core.storage.ResourceMeta
+import com.dtolabs.rundeck.core.storage.StorageTree
+import groovy.transform.CompileStatic
+import groovy.util.logging.Slf4j
+import org.rundeck.storage.api.Resource
+/**
+ * Iterates over given storage tree, and updates resources according to the config
+ */
+@CompileStatic
+@Slf4j
+class TreeUpdater implements Updater {
+    StorageTree storageTree
+
+    Boolean enabled
```

Comment:
I feel thse two configuration parameters better belong to the `UpdaterConfig` subclass: `JasyptEncryptionEnforcerUpdaterConfig`. 

```
public JasyptEncryptionEnforcerUpdaterConfig implements UpdaterConfig {
    Boolean enabled
    String basePath

    @Override
    boolean shouldPerform() {
       if (!enabled) {
            return false
        }

        def configuration = treeCreator.storageConfigMap
        return configuration.entrySet().find {
            it.key =~ /^converter\.\d+\.type/ && it.value == 'jasypt-encryption'
        }
    }
}
```

This way, we can separate the concern on config from the TreeUpdater logic which can be stateless